test/GPv2Trade/ExtractOrder.t.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {GPv2Order, IERC20} from "src/contracts/libraries/GPv2Trade.sol";
+import {GPv2Signing} from "src/contracts/mixins/GPv2Signing.sol";
+
+import {Helper} from "./Helper.sol";
+
+import {Order as OrderLib} from "test/libraries/Order.sol";
+import {SettlementEncoder} from "test/libraries/encoders/SettlementEncoder.sol";
+
+contract ExtractOrder is Helper {
+    using SettlementEncoder for SettlementEncoder.State;
+
+    struct Fuzzed {
+        address sellToken;
+        address buyToken;
+        address receiver;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint32 validTo;
+        bytes32 appData;
+        uint256 feeAmount;
+        uint256 executedAmount;
+    }
+
+    function sampleOrder() private returns (GPv2Order.Data memory order) {
+        order = GPv2Order.Data({
+            sellToken: IERC20(makeAddr("GPv2Trade.ExtractOrder sampleOrder sell token")),
+            buyToken: IERC20(makeAddr("GPv2Trade.ExtractOrder sampleOrder buy token")),
+            receiver: makeAddr("GPv2Trade.ExtractOrder sampleOrder receiver"),
+            sellAmount: 42 ether,
+            buyAmount: 13.37 ether,
+            validTo: type(uint32).max,
+            appData: bytes32(0),
+            feeAmount: 1 ether,
+            kind: GPv2Order.KIND_SELL,
+            partiallyFillable: false,
+            sellTokenBalance: GPv2Order.BALANCE_ERC20,
+            buyTokenBalance: GPv2Order.BALANCE_ERC20
+        });
+    }
+
+    function assertSameOrder(GPv2Order.Data memory lhs, GPv2Order.Data memory rhs) private pure {
+        uint256 assertionCount = 12;
+        assertEq(address(lhs.sellToken), address(rhs.sellToken));
+        assertEq(address(lhs.buyToken), address(rhs.buyToken));
+        assertEq(lhs.receiver, rhs.receiver);
+        assertEq(lhs.sellAmount, rhs.sellAmount);
+        assertEq(lhs.buyAmount, rhs.buyAmount);
+        assertEq(lhs.validTo, rhs.validTo);
+        assertEq(lhs.appData, rhs.appData);
+        assertEq(lhs.feeAmount, rhs.feeAmount);
+        assertEq(lhs.kind, rhs.kind);
+        assertEq(lhs.partiallyFillable, rhs.partiallyFillable);
+        assertEq(lhs.sellTokenBalance, rhs.sellTokenBalance);
+        assertEq(lhs.buyTokenBalance, rhs.buyTokenBalance);
```

Comment:
Can we not simply do `assertEq(keccak256(abi.encode(lhs)), keccak256(abi.encode(rhs)))` ?

Reply:
Right! Fixed, 16 lines less. :slightly_smiling_face: